SwanFileBrowser/src/listing.ts

```diff
@@ -0,0 +1,2390 @@
+/**
+ * File took from https://github.com/jupyterlab/jupyterlab/blob/master/packages/filebrowser/src/listing.ts
+ * for jupytrelab version 3.0.x.
+ *
+ * In this file, we reimplement the DirListing class in the SwanDirListing class.
+ * This allows you to manipulate the low-level entries in the file browser widget,
+ * allowing us, for example, to change the icon if the folder is a project.
```

Comment:
Same comment as above: this is a huge file, it would be hard for a developer to figure out what you changed in the upstream file when moving to a newer version of this. As we discussed via zoom, I'm concerned about the maintainability of this extension. It would make it easier to have some documentation that points to the key modifications of the original file in detail.

Reply:
I did it, let me know if this is good enough for you. :D